pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -464,12 +471,17 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings)
 
 func setUpstreamProtocol(cluster *v2.Cluster, port *model.Port) {
 	if port.Protocol.IsHTTP2() {
-		cluster.Http2ProtocolOptions = &core.Http2ProtocolOptions{
-			// Envoy default value of 100 is too low for data path.
-			MaxConcurrentStreams: &types.UInt32Value{
-				Value: 1073741824,
-			},
-		}
+		designateH2(cluster)
```

Comment:
I suggest a simpler way: get rid of the if port is http2 check. And in the callsite, check for if port is http2 or tlsmode is istio_mutual, setUpstreamH2..

Reply:
There are 4 calls to setUpstreamProtocol(), which as far as the name goes is generic.
It will be unwieldy. there is also a PR coming for h2 upgrade for not mtls, which will need this `designateH2` function as well.